js/capmap.js

```diff
@@ -269,61 +269,48 @@ function changeLanguage() {
 }
 
 function buildLegend() {
-  // rebuild legend if useMinorThreat = true
+  // Rebuild legend if useMinorThreat is true
   if (alertOptions.useMinorThreat) {
-    var div = document.getElementById('legend-warning-types')
-    div.innerHTML = ''
-
-    var tr = document.createElement('tr')
-
-    var td = document.createElement('td')
-    td.className = 'colorLegend'
-    td.id = 'levelGreen'
-    tr.appendChild(td)
-
-    td = document.createElement('td')
-    td.id = 'levelGreenText'
-    td.innerHTML = "minor thread";
-    tr.appendChild(td)
-
-    td = document.createElement('td')
-    td.className = 'colorLegend'
-    td.id = 'levelOrange'
-    tr.appendChild(td)
-
-    td = document.createElement('td')
-    td.id = 'levelOrangeText'
-    td.innerHTML = "dangerous";
-    tr.appendChild(td)
-
-    div.appendChild(tr)
-
-    tr = document.createElement('tr')
-
-    var td = document.createElement('td')
-    td.className = 'colorLegend'
-    td.id = 'levelYellow'
-    tr.appendChild(td)
-
-    td = document.createElement('td')
-    td.id = 'levelYellowText'
-    td.innerHTML = "potentially dangerous";
-    tr.appendChild(td)
-
-    td = document.createElement('td')
-    td.className = 'colorLegend'
-    td.id = 'levelRed'
-    tr.appendChild(td)
-
-    td = document.createElement('td')
-    td.id = 'levelRedText'
-    td.innerHTML = "very dangerous";
-    tr.appendChild(td)
+    var div = document.getElementById('legend-warning-types');
+    div.innerHTML = ''; // Clear existing content
+
+    // Legend data
+    var levels = [
+      { id: 'levelGreen', textId: 'levelGreenText', text: 'minor threat' },
+      { id: 'levelOrange', textId: 'levelOrangeText', text: 'dangerous' },
+      { id: 'levelYellow', textId: 'levelYellowText', text: 'potentially dangerous' },
+      { id: 'levelRed', textId: 'levelRedText', text: 'very dangerous' }
+    ];
+
+    // Create table rows for the legend
+    for (let i = 0; i < levels.length; i += 2) {
```

Comment:
(With my not-so-up-to-date skills), is it this syntax for incrementing OK, and is it incremented by 2 on purpose?

Reply:
My bad. Yes it is on purpose. After reading the code :)

Reply:
Yeah, in this case, however, +1 would give the same result